### PR TITLE
Don't sync IPtables before underlying store/reflector is fully synced

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #42108

Build on top of #42108 - only the second commit is unique.

### CODE REVIEW

File: pkg/proxy/config/config.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ const (
 	ADD Operation = iota
 	UPDATE
 	REMOVE
+	INITIAL_SYNC_DONE
```
Comment (smarterclayton) [inline pkg/proxy/config/config.go]: SYNCED?

**Comment by smarterclayton** [comment]
Code looks correct to me.

**Comment by smarterclayton** [comment]
golint

```
W0227 09:03:39.386] pkg/proxy/config/config.go:190:1: exported method ServiceConfig.RegisterHandler should have comment or be unexported
W0227 09:03:39.386] pkg/proxy/config/config.go:197:1: exported method ServiceConfig.Channel should have comment or be unexported
W0227 09:03:39.387] pkg/proxy/config/config.go:208:1: exported method ServiceConfig.Config should have comment or be unexported
W0227 09:03:39.387] 
W0227 09:03:39.387] Please fix the above errors. You can test via "golint" and commit the result.
W0227 09:03:39.387] 
I0227 09:03:39.487] FAILED   hack/make-rules/../../hack/verify-golint.sh	35s
```

**Comment by wojtek-t** [comment]
@smarterclayton - should be fixed now.

**Comment by smarterclayton** [comment]
@k8s-bot non-cri e2e test this

**Comment by thockin** [comment]
Looks good, just waiting for dep to clear, I guess?

**Comment by wojtek-t** [comment]
> Looks good, just waiting for dep to clear, I guess?

Yes

**Comment by wojtek-t** [comment]
@thockin @smarterclayton - the base PR has just merged. Can you please lgtm this one now?

**Pull request merged**